datafusion/core/src/execution/context/mod.rs

```diff
@@ -1196,9 +1184,11 @@ impl SessionContext {
 
             match function_factory {
                 Some(f) => f.create(&state, stmt).await?,
-                _ => Err(DataFusionError::Configuration(
-                    "Function factory has not been configured".into(),
-                ))?,
+                _ => {
+                    return Err(DataFusionError::Configuration(
+                        "Function factory has not been configured".to_string(),
+                    ))
+                }
```

Comment:
I think we can use `config_err!` here.
```suggestion
                _ => {
                    return config_err!("Function factory has not been configured")
                }
```